internal/pbkdf2/pbkdf2.go

```diff
@@ -0,0 +1,145 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package pbkdf2
+
+import (
+	"crypto"
+	"errors"
+	"math"
+	"time"
+
+	"golang.org/x/crypto/pbkdf2"
+)
+
+const (
+	benchmarkPassword = "foo"
+)
+
+var (
+	benchmarkSalt = []byte("0123456789abcdefghijklmnopqrstuv")
+)
+
+var timeExecution = func(params *Params) time.Duration {
+	start := time.Now()
+	if _, err := Key(benchmarkPassword, benchmarkSalt, params, uint(params.HashAlg.Size())); err != nil {
+		panic(err)
+	}
+	return time.Now().Sub(start)
+}
```

Comment:
this bit of code in itself is not reached by tests

Reply:
I've added a test that doesn't mock this now.